packages/dataviews/src/normalize-form-fields.ts

```diff
@@ -1,42 +1,50 @@
 /**
  * Internal dependencies
  */
+import type { Layout, LayoutType } from './layout-types';
 import type { Form } from './types';
 
 interface NormalizedFormField {
 	id: string;
-	layout: 'regular' | 'panel';
-	labelPosition: 'side' | 'top' | 'none';
+	layout: Layout;
 }
 
 export default function normalizeFormFields(
 	form: Form
 ): NormalizedFormField[] {
-	let layout: 'regular' | 'panel' = 'regular';
-	if ( [ 'regular', 'panel' ].includes( form.type ?? '' ) ) {
-		layout = form.type as 'regular' | 'panel';
+	let layout: LayoutType = 'regular';
+	const formLayout = form.layout ?? {
+		type: 'regular',
+		labelPosition: 'top',
+	};
+	if ( [ 'regular', 'panel', 'card' ].includes( formLayout.type ?? '' ) ) {
+		layout = formLayout.type as 'regular' | 'panel' | 'card';
 	}
 
 	const labelPosition =
-		form.labelPosition ?? ( layout === 'regular' ? 'top' : 'side' );
+		formLayout.labelPosition ?? ( layout === 'regular' ? 'top' : 'side' );
 
 	return ( form.fields ?? [] ).map( ( field ) => {
 		if ( typeof field === 'string' ) {
 			return {
 				id: field,
-				layout,
-				labelPosition,
+				layout: {
+					type: layout,
+					labelPosition,
+				} as Layout,
```

Comment:
Removing this cast would give us a better type system. I think that, for now, we can keep this approach and improve the type system in another PR.